### PR TITLE
treat multiple name entries with the same email as one author

### PR DESCRIPTION
**Pull request by jtmr05**
Sometimes, there are multiple entries (as per `git shortlog -nse`) with the same email but different names. This results in them being treated as separate authors when in reality they're not. With this change, authors are grouped according to their email address and the displayed name is the one that occurs the most. 

Disclaimer: I'm Rust newbie, therefore code may be far from optimal.

### CODE REVIEW

File: src/info/utils/git.rs
Comment (spenserblack) [inline src/info/utils/git.rs:105]: This is fine with me, but I think it's a bit more idiomatic to write something more functional, for example using `Iterator::fold` or `Iterator::max_by`, and also `Iterator::collect`.
```rust
let authors_by_number_of_commits = email_to_names.into_iter().map(|(email, name_to_commits)| {
    let most_frequent_name = name_to_commits.into_iter().max_by(|(name, commits) {
        // implementation
        // NOTE: You may need to use `.fold` instead to reduce to one iteration.
    });
    Sig { /* ... */ }; // Now this is an iterator of Sigs that can be collected
}).collect();
```
IIRC the Rust compiler is pretty good about unfolding iterators into something more efficient, though I may be wrong.

I haven't used Rust extensively in a while (should I say I've gotten rusty? :laughing:) so some of this may be incorrect.

**Review by spenserblack — COMMENTED** [review]
:thinking: This is definitely a cool idea. Don't really have strong opinions either way, personally.

Previously, we relied on `.mailmap`, but this PR makes things work more "out of the box." :+1:

However, just from browsing the GitHub community discussions, I've seen a lot of confusion where the name is the same but the *email is different.* Many users expect the name, not the email, to be the identifier. IMO email makes much more sense, but I'm just throwing it out there that some may find it confusing. This often happens, for example, when commits made in the GitHub web UI use the "do not reply" email.

File: src/info/utils/git.rs
Comment (jtmr05) [reply src/info/utils/git.rs:32]: hmm, I didn't touch this at all

File: src/info/utils/git.rs
Comment (spenserblack) [inline src/info/utils/git.rs:107]: I was thinking something kind of like this. Might be some mistakes below. Writing more than a single line in a GitHub suggestion block is a pain :laughing:

Doesn't have to be exactly like this, but destructuring can really help make things more readable than `.0` and `.1`. And if you're mutating something inside of a `.fold`, there's a decent chance that it can be part of the fold.
```suggestion
                let (most_frequent_name, author_nbr_of_commits) = name_to_nbr_of_commits
                    .into_iter()
                    .fold((None, 0), |(common_name, total), (name, commits)| {
                        let common_name = match common_name {
                            Some((n, count) if count > commits => common_name,
                            _ => Some((name, commits)),
                        };
                                
                        (common_name, total + commits)
                    })
                    .unwrap();
```

File: src/info/utils/git.rs
Comment (spenserblack) [reply src/info/utils/git.rs:32]: :thinking: Well, I'm a bit obsessive about clean git histories, so I'll probably change this back before merging 🙂

**Review by spenserblack — APPROVED** [review]
Alright, thanks!

As far as code goes, LGTM.

Behavior-wise, we'll definitely want @o2sh, and maybe some other maintainers, to offer their feedback.

**Comment by o2sh** [comment]
This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

**Comment by jtmr05** [comment]
what? you haven't reviewed it yet...

**Comment by o2sh** [comment]
Apologies, @jtmr05, for the delay in reviewing the PR. 

We've already addressed this issue in the past #447 and decided to stop identifying authors by email and stick to what git shortlog does - as the de facto standard - and merge entries based on `username + email` by default with a `.mailmap` mapping if present.

**Comment by jtmr05** [comment]
that's reasonable, I feel like this approach could at least be included as an option

**Comment by spenserblack** [comment]
>what? you haven't reviewed it yet...

For future reference, that's an automated message, so it doesn't necessarily mean we're dismissing your PR/issue 🙂 

>I feel like this approach could at least be included as an option

I feel like `.mailmap` kind of is that option, and preferable since git itself will also use it to identify authors. In other words, if get duplicate authors from onefetch, you probably need to set a `.mailmap` to get your authorship sorted out for other tools, too.

This does make me wonder if we could suggest a script or something to help generate a `.mailmap` file to help recreate your preferred behavior :thinking:

**Review by spenserblack — CHANGES_REQUESTED** [review]
Undoing my approval since

- We probably want to put more thought into how to handle this before merging
- There are merge conflicts, anyway :wink:

**Comment by o2sh** [comment]
This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

**Pull request closed without merging**
